examples/example_02_matlab_function/treeOpt_matlab_example.py

```diff
@@ -0,0 +1,37 @@
+from treeopt.treeOpt import adaptive_metamodell
+from treeopt import simulate
+import numpy as np
+
+
+def calc_matlab(x):
+    inputFile = "input.txt"
+    outputFile = "output.txt"
+    simFile = "matlab_function.m"
+    program = "matlab"
+    responce = simulate.simulate_external_programm(
+        inputFile, outputFile, simFile, program, x
```

Comment:
The function in line 11 starts a python-subprocess, which starts the program and the file which is to be executed. When the subprocess is finished, it reads the output file and outputs its value. I have not tried this function with a lot of programs, but as long as the program allows for it to be started in a shell, with a file, this method should work.

Reply:
Yes, but you have very explicit requirements on what the input file and output file have to look like, so that you can write to it and read from it. That means that in 99% of the cases, a user of your library cannot call the external program directly, but they have to write a script that converts your required numpy-conform input and output files to the input and output files of the external program. Think of abaqus for instance.  Or did I misunderstand your code?

If I understand your code correctly, than here is what I suggest: Because the user of your library *must* tell your code how to set the parameters to the external program and how to read the response from any generated output files, why don't you change your API in such a way, that the user must supply three functions `set_parameters`, `start_program` and `read_response` to call external programs. Then the converson can be performed in the `set_parameters` and `read_response` functions by the user. Your library does not have to know how the actual input and output files of the external program look like and it does not have to make any assumptions about it.